File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/blockout-date.model.ts
import { SystemRoleEnum, UserInfoModel } from '../../share/models/user-info.model';

export enum BlockoutDateStatus {
  Draft = 'Draft',
  Active = 'Active'
}
export interface IBlockoutDateModel {
  id?: string;
  title: string;
  description: string;
  createdBy: string;
  startDay: number;
  startMonth: number;
  endDay: number;
  endMonth: number;
  validYear: number;
  status?: BlockoutDateStatus;
  planningCycleId?: string;
  serviceSchemes: string[];
  isConfirmed: boolean;
}

export class BlockoutDateModel implements IBlockoutDateModel {
  public id?: string;
  public title: string = '';
  public description: string = '';
  public createdBy: string = '';
  public startDay: number = 0;
  public startMonth: number = 0;
  public endDay: number = 0;
  public endMonth: number = 0;
  public validYear: number = 0;
  public status?: BlockoutDateStatus = BlockoutDateStatus.Draft;
  public planningCycleId?: string = '';
  public serviceSchemes: string[] = [];
  public isConfirmed: boolean = false;

  public static haveCudPermission(user: UserInfoModel): boolean {
    return user.hasRole(SystemRoleEnum.SystemAdministrator) || user.hasRole(SystemRoleEnum.CoursePlanningCoordinator);
  }

  constructor(data?: IBlockoutDateModel) {
    if (data) {
      this.id = data.id;
      this.title = data.title;
      this.description = data.description;
      this.createdBy = data.createdBy;
      this.startDay = data.startDay;
      this.startMonth = data.startMonth;
      this.endDay = data.endDay;
      this.endMonth = data.endMonth;
      this.validYear = data.validYear;
      this.status = data.status;
      this.planningCycleId = data.planningCycleId;
      this.serviceSchemes = data.serviceSchemes;
      this.isConfirmed = data.isConfirmed;
    }
  }
}

File: pdlearning/src/microservices/course-microservice/Api/Application/Models/BlockoutDateModel.cs
using System;
using System.Collections.Generic;
using Microservice.Course.Domain.Entities;
using Microservice.Course.Domain.Enums;

namespace Microservice.Course.Application.Models
{
    public class BlockoutDateModel
    {
        public BlockoutDateModel()
        {
        }

        public BlockoutDateModel(BlockoutDate entity)
        {
            Id = entity.Id;
            CreatedBy = entity.CreatedBy;
            StartDay = entity.StartDay;
            StartMonth = entity.StartMonth;
            EndDay = entity.EndDay;
            EndMonth = entity.EndMonth;
            PlanningCycleId = entity.PlanningCycleId;
            ValidYear = entity.ValidYear;
            ServiceSchemes = entity.ServiceSchemes;
            Title = entity.Title;
            Description = entity.Description;
            Status = entity.Status;
            IsConfirmed = entity.IsConfirmed;
        }

        public string Title { get; set; }

        public string Description { get; set; }

        public BlockoutDateStatus Status { get; set; }

        public Guid Id { get; set; }

        public Guid? CreatedBy { get; set; }

        public int StartDay { get; set; }

        public int StartMonth { get; set; }

        public int EndDay { get; set; }

        public int EndMonth { get; set; }

        public int ValidYear { get; set; }

        public Guid PlanningCycleId { get; set; }

        public bool IsConfirmed { get; set; }

        public IEnumerable<string> ServiceSchemes { get; set; } = new List<string>();
    }
}
